src/backends/torch/torchlib.cc

```diff
@@ -1369,6 +1375,10 @@ namespace dd
   int TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy,
                TMLModel>::predict(const APIData &ad_in, APIData &out)
   {
+    if (!_concurrent_predict) {
+        // concurrent calls can use more memory on gpu than initially expected
+        std::lock_guard<std::mutex> lock(_net_mutex);
```

Comment:
Since lock_guard is RAII, it should be destroyed at the end of the `if`, hence not preventing concurrent execution? Unless I'm missing something

Reply:
You could replace it by a unique_ptr scoped to the whole function